app/controllers/home_controller.rb

```diff
@@ -1,5 +1,125 @@
 class HomeController < ApplicationController
   before_action :authenticate_user!
   def index
+    @id = current_user.email
   end
+
+  def getYears
+    sems = Course.where(teacher:@id)
+    uniqSems = Set.new
+    sems.each do |s|
+      uniqSems << s.semester[-4..-1]
+    end
+    return uniqSems.length()
+  end
+  helper_method :getYears
+
+
+  def getAvgTotal
+    avg = Quiz.where(teacher:@id).average(:score)
+    if avg.nil?
+      avg = "No quizzes taken!"
+    else 
+      avg = avg.round(2)
+    end
+
+    strk = Quiz.where(teacher:@id).average(:longest_streak)
+    if strk.nil?
+      strk = "No quizzes taken!"
+    else
+      strk = strk.round(2)
+    end
+
+    return [avg, strk]
+  end
+  helper_method :getAvgTotal
+
+  def getAvgRecent
+    recentFall = Course.where(teacher:@id).where("semester like ?", "%#{"Fall"}%").order(semester: :desc).first
+    fallYear = (recentFall.semester[-4..-1] || "0").to_i
+    recentSpring = Course.where(teacher:@id).where("semester like ?", "%#{"Spring"}%").order(semester: :desc).first
+    springYear = (recentSpring.semester[-4..-1] || "0").to_i
+   
+    semester = ""
+    if fallYear > springYear
+      semester = recentFall.semester
+    else
+      semester = recentSpring.semester
+    end
+   
+    courses = Course.select(:id).where(teacher:@id).where("semester like ?", "%#{semester}%")
+    
+    avg = Quiz.where(teacher:@id,id:courses).average(:score)
+    if avg.nil?
+      avg = "No quizzes taken!"
+    else 
+      avg = avg.round(2)
+    end
+
+    strk = Quiz.where(teacher:@id,id:courses).average(:longest_streak)
+    if strk.nil?
+      strk = "No quizzes taken!"
+    else
+      strk = strk.round(2)
+    end
+    
+    return [semester, avg, strk]
+  end
+  helper_method :getAvgRecent
+
+  def studentInfo
+    qID = Quiz.where(teacher:@id).pluck(:id)
+    atm = Qroster.where(quiz_id:qID,correct_resp:true).count(:attempts)
+
+    best = Qroster.where(quiz_id:qID,correct_resp:true).group(:student_id).select(:student_id, "(SUM(CAST(1 AS Float) / CAST(attempts as Float)*100.00) / #{atm}) AS wscore", "(SUM(CAST(1 AS Float) / CAST(attempts as Float)*100.00)/COUNT(attempts)) AS score").order("wscore DESC").first
```

Comment:
`(SUM(CAST(1 AS Float) / CAST(attempts as Float)*100.00) / #{atm}` ---> (SUM(CAST(1 AS Float) + CAST(attempts as Float)*100.00) / #{atm}?

Based on my understanding, I could be wrong.

Reply:
also, what's the difference b/w wscore and score? looks like the same formula

Reply:
The original equation should be right because I'm looking at the ratio of 1 correct attempt to the attempts made, so 1/attempts

wscore is a "weighted average", so it's the score formula * (# of attempts for that student)/(# of all attempts for that profs quizzes)

But the (# of attempts for that student) part cancels